src/modules/appUI.js

```diff
@@ -12,16 +36,31 @@ const filmTemplate = (info) => `
           <i class="far fa-heart"></i>
         </div>
       </div>
-      <button type="button" class="comments">Comments</button>
+      <button type="button" class="comments" id="${info.id}">Comments</button>
     </li>
   `;
-
-const displayMovies = () => {
+const displayMovies = async () => {
   const filmsContainer = document.querySelector('.films');
+
   filmsContainer.innerHTML = '';
-  getMovies().forEach(async (movie) => {
+  const movieList = getMovies().map(async (movie, index) => {
     const movieData = await movie;
     filmsContainer.innerHTML += filmTemplate(movieData);
+    const btnComment = document.querySelectorAll('.comments');
+    btnComment.forEach((x) => {
+      x.addEventListener('click', async () => {
+        const resp = await fetch(
+          `https://api.tvmaze.com/lookup/shows?thetvdb=${x.id}`
+        );
+        const data = await resp.json();
```

Comment:
I think it would be better to avoid fetching the movies data again since we already have a function `getMovies()`  doing that. I have noted that this second call to the API is delaying the display of the popup image.
```suggestion
        const resp = getMovies()[index] // where index = the clicked button's id;
        const data = await resp.json(); 
```